rpc/gnmi/gnmi-config-subscriptions.md

```diff
@@ -0,0 +1,181 @@
+which makes this RPC rather ineffective on YANG schemas that do not employ
+a separation of config and state elements by using distinct containers.
+
+This proposal introduces the Config Subscription extension that allows clients
+to indicate to servers that they are interested in configuration values only.
+
+gNMI Config Subscription Extension proto specification is defined in
+[gnmi_ext.proto](
+https://github.com/openconfig/gnmi/blob/master/proto/gnmi_ext/gnmi_ext.proto
+).
+
+## 2. Definition
+
+A `ConfigSubscription` message is embedded as an extension message in the
+`SubscribeRequest` or `SubscribeResponse` proto messages.
+If the extension is embedded in a `SubscribeRequest`, the action field
+must be a `ConfigSubscriptionStart`.
+The presence of such an extension indicates to the target that the client
+wants to start a ConfigSubscription. The target must return notifications
+pertaining to data leaves that the target considers to be writable.
+If the subscription type is `ON_CHANGE`, the target must separate the
+notifications triggered by different commits using a
+`ConfigSubscriptionSyncDone` in a a `SubscribeResponse` message.
+On the other hand, if it is embedded in a `SubscribeResponse`, the action
+field must be a `ConfigSubscriptionSyncDone`. This action is used by a
+target to indicate a commit boundary to the client.
+
+## 2.1 Proto
+
+The extension contains a message called `ConfigSubscription` that carries
+one of 2 types of actions. `ConfigSubscriptionStart` or
+`ConfigSubscriptionSyncDone`
+
+```proto
+// ConfigSubscription extension allows clients to subscribe to configuration
+// schema nodes only.
+message ConfigSubscription {
+  oneof action {
+    // ConfigSubscriptionStart is sent by the client in the SubscribeRequest
+    ConfigSubscriptionStart start = 1;
+    // ConfigSubscriptionSyncDone is sent by the server in the SubscribeResponse
+    ConfigSubscriptionSyncDone sync_done = 2;
+  }
+}
+
+// ConfigSubscriptionStart is used to indicate to a target that for a given set
+// of paths in the SubscribeRequest, the client wishes to receive updates
+// for the configuration schema nodes only.
+message ConfigSubscriptionStart {}
+
+// ConfigSubscriptionSyncDone is sent by the server in the SubscribeResponse
+// after all the updates for the configuration schema nodes have been sent.
+message ConfigSubscriptionSyncDone {
+  // ID of a commit confirm operation as assigned by the client
+  // see Commit Confirm extension for more details.
+  string commit_confirm_id = 1;
+  // ID of a commit as might be assigned by the server
+  // when registering a commit operation.
+  string server_commit_id = 2;
+}
+```
+
+## 2.2 Actions
+
+### 2.2.1 ConfigSubscriptionStart
+
+A `ConfigSubscriptionStart` message is used by a gNMI client in a
+`SubscribeRequest` to indicate that it wants to start a ConfigSubscription.
+The target must respond exclusively with configuration data relevant to the
```

Comment:
I think it would be good to include an explicit definition if a notification is expected to be sent with a subscribe response for P1.  Following the pattern of subscribe, I am thinking the target should send a subscribe response with the path(s) requested, even in an ON_CHANGE subscription.

Reply:
I added the following paragraph for clarification:

The target must respond with an initial set of updates followed by a
`SubscribeResponse` where the `sync_update` field is set to `true`.
However, if the `updates_only` field in the `SubscribeRequest` was set to
`true`, the target should skip sending the initial updates and only send
a `SubscribeResponse` with `sync_update` set to `true` as per the base gNMI
specification.

Reply:
thank you!